app/models/manageiq/providers/kubevirt/infra_manager/provision/cloning.rb

```diff
@@ -26,18 +25,43 @@ def log_clone_options(clone_options)
   end
 
   def start_clone(options)
-    # Retrieve the details of the source template:
-    template = nil
-    source.with_provider_connection(:namespace => source.location) do |connection|
-      template = connection.template(source.name)
-      template.clone(user_options(options))
+    source.with_provider_connection do |connection|
+      # Retrieve the details of the source template:
+      template    = source.provider_object
+      namespace ||= source.location
 
-      vm = connection.vm(options[:name])
+      params    = values(template, user_options(options))
+      pvc_specs = template.objects.select { |o| o[:kind] == "PersistentVolumeClaim" }&.map(&:to_h)
+      vm_spec   = template.objects.detect { |o| o[:kind] == "VirtualMachine" }&.to_h
+      raise N_("No Virtual Machine defined in template") if vm_spec.nil?
 
-      phase_context[:new_vm_ems_ref] = vm.uid
+      param_substitution!(vm_spec, params)
```

Comment:
Is this a new feature? I don't see it in any of the deleted code, and it didn't do any of this previously.